pkg/controller/association/reconciler.go

```diff
@@ -263,7 +267,13 @@ func (r *Reconciler) reconcileAssociation(ctx context.Context, association commo
 
 	url, err := r.AssociationInfo.ExternalServiceURL(r.Client, association)
 	if err != nil {
-		return commonv1.AssociationPending, err // maybe not created yet
+		// don't log an error if the service is not found and its parent resource has just been created
+		timeSinceCreation := time.Since(association.Associated().GetCreationTimestamp().Time)
+		maybeNotYetCreated := timeSinceCreation < timeToExpectServiceAfterCreation
+		if apierrors.IsNotFound(err) && maybeNotYetCreated {
+			return commonv1.AssociationPending, nil
+		}
```

Comment:
If the `Service` does not exist yet, and if we think that it can be confusing for the user to see an error, should we just log this event at the info level, and maybe generate a warn event? If the `Service` cannot be created I would expect the error to be logged by the relevant controller anyway.

```suggestion
		if apierrors.IsNotFound(err) {
			// Service is expected to be created by the resource controller
			k8s.EmitErrorEvent(r.recorder, err, association, events.EventAssociationError,
				"Associated resource Service for %s does not exist yet: %v", assocRef.NamespacedName(), err)
			log.Info("Associated resource Service is not available yet", "error", err, "name", association.Associated().GetName(), "ref_name", assocRef.Name)
			return commonv1.AssociationPending, nil
		}
```

Reply:
You're right, the resource controller is doing its job and will report any errors. I'm good at just logging the event at the info level but I'm not in favor of generating a warn event. Otherwise, I feel like that we are moving the noise from the operator log to the k8s event stream.

Reply:
You're right, the resource controller is doing its job and will report any errors. I'm good at just logging the event at the info level but I'm not in favor of generating a warn event. Otherwise I feel like we're moving the noise from the operator log to the k8s event stream.